source/customization/menus.rst

```diff
@@ -1,5 +1,367 @@
+
+
+.. _menu-types:
+
+Available Menus
+===============
+
+The file name
+of a ``.sublime-menu`` file
+specifies the menu to be modified
+in the application.
+
+The following menus are available:
+
+========================   ===============
+     File/Menu Name          Description
+========================   ===============
+**Main**                   Main menu
+
+**Context**                Context menu in the editing area
+
+**Find in Files**          Appears when clicking the "…" button
+                           in the :doc:`Find in Files
+                           </search_and_replace/search_and_replace_files>` panel
+
+**Side Bar**               Context menu for each node in the sidebar
+
+**Side Bar Mount Point**   Additional context menu items
+                           for the top-level nodes in the sidebar
+
+**Tab Context**            Context menu of the tab bar
+
+**Widget Context**         Context menu of input fields
+                           in all kinds of widgets,
+                           including Command Palette, Goto Anything,
+                           the Find panels
+                           and panels opened by plugins
+========================   ===============
+
+Additionally,
+the following four menus open
+when you click the respective section
+in the status bar:
+
+- **Encoding**
+- **Line Endings**
+- **Indentation**
+- **Syntax**
+
+.. figure:: images/statusbar_menu.gif
+   :class: shadowed
+
+   Demonstration of a status bar menu.
+
+
+.. _menu-items:
+
+Menu Items
+==========
+
+A menu item can either
+invoke a command (with arguments)
+when it is selected,
+or have a submenu.
+
+The available properties are:
+
+- a command name,
+- arguments for the command,
+- an ID,
+- a caption,
+- a mnemonic and
+- a submenu.
+
+In order to function properly,
+a menu item must provide at least:
+
+- a command name,
+- a caption and a submenu,
+- just a caption, or
+- an ID (see :ref:`below <item-ids>`).
+
+When parsing a menu item,
+the following rules apply:
+
+#. A menu item with a submenu
+   cannot invoke command.
+   It also overrides the :ref:`separator caption <menu-separators>`
+   and causes the item to render
+   a literal hyphen as caption.
+
+#. If no caption is provided,
+   a caption is inferred
+   from the command's ``description`` method.
+   If neither caption nor command are provided,
+   the caption will be an empty string.
+
+#. The character used for the mnemonic
+   must be contained in the item's caption.
+   Mnemonics are case-sensitive.
+
+#. Menu items referencing
+   commands that cannot be found
+   are disabled.
+
+#. Menu items can be hidden or disabled
+   by their referenced command.
+
+.. TODO add refs to Command methods
+
+
+.. _menu-separators:
+
+Separators
+**********
+
+Separators are menu items
+with the caption ``-``
+and no submenu.
+They are commonly used
+to group menu items with similar purposes
+or that are otherwise related.

```

Comment:
I initially had it like that, but the flow of the sentence seems off because it is not clear  we are referring to "menu items" since they are used differently in the sentence before.